great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py

```diff
@@ -14,36 +14,48 @@
 class DefaultExpectationConfigurationBuilder(ExpectationConfigurationBuilder):
     """
     Class which creates ExpectationConfiguration out of a given Expectation type and
-    parameter_name-to-parameter_fully_qualified_parameter_name map (name-value pairs supplied as kwargs dictionary).
+    parameter_name-to-parameter_fully_qualified_parameter_name map (name-value pairs supplied as expectation_kwargs and
+    kwargs dictionaries).
     """
 
-    def __init__(self, expectation_type: str = None, **kwargs):
+    def __init__(
+        self,
+        expectation_type: str,
+        expectation_kwargs: Optional[Dict[str, str]] = None,
+        meta: Optional[Dict[str, Any]] = None,
+        success_on_last_run: Optional[bool] = None,
+        **kwargs,
+    ):
         self._expectation_type = expectation_type
-        self._parameter_name_to_fully_qualified_parameter_name_dict = kwargs
+        if expectation_kwargs is None:
+            expectation_kwargs = {}
+        self._expectation_kwargs = expectation_kwargs
+        self._expectation_kwargs.update(kwargs)
+        if meta is None:
+            meta = {}
```

Comment:
Yes @talagluck ! It's one of the Python gotchas (I learned from experience by screwing this up). If you instantiate the dict on line 25, it will be created when the class is first initialized, then every instance will mutate the same dict (likely will lead to strange errors). Here we are creating the dict in the constructor, e.g. one dict for each instance.

Reply:
Ah - makes sense! I think I remember discussing this with you before, but I must not have fully processed it.